python/tk_framework_adobe/rpc/communicator.py

```diff
@@ -565,9 +567,9 @@ def _handle_response(self, response, *args):
         self.log_network_debug("Response UID is %s" % uid)
 
         try:
-            self._RESULTS[uid] = self._ensure_utf8(json.loads(result["result"]))
+            self._RESULTS[uid] = json.loads(six.ensure_str(result["result"]))
         except (TypeError, ValueError):
-            self._RESULTS[uid] = self._ensure_utf8(result.get("result"))
+            self._RESULTS[uid] = six.ensure_str(result.get("result"))
```

Comment:
I've replaced this, though I'm not convinced this fallback section would be useful because it expects to get a dictionary as far as I can see, so a string would cause it to break.

Reply:
I tested deliberately adding a value error to the try before so this code got run and it errored as I expected. I tried without the `ensure_str` as well, just in case I was converting it, but it still failed as it was gathering a `str` object. So I'm not sure what this code is/was trying to do.

Reply:
I wonder if there is some edge case this is trying to handle.

Reply:
I've added a TODO, and tried to more closely match the original behavior, though I still suspect this code is not right.